app/models/manageiq/providers/inventory/persister/builder/physical_infra_manager.rb

```diff
@@ -48,9 +48,33 @@ def physical_storages
           add_common_default_values
         end
 
-        def canisters
+        def physical_switches
+          add_properties(:manager_ref => %i(uid_ems))
           add_common_default_values
         end
+
+        # TODO! serial_number wasn't in old refresh's find_key
+        # but there can be more canisters for one storage
+        # http://sysmgt.lenovofiles.com/help/index.jsp?topic=%2Fcom.lenovo.lxca_restapis.doc%2Frest_apis_reference.html
+        # In doc, there is UUID, but not used in this provider
+        # # Tmp solution with serial_number works for specs
+        def canisters
+          add_properties(:manager_ref => %i(physical_storage serial_number))
+          add_properties(:manager_ref_allowed_nil => %i(serial_number))
```

Comment:
I'm concerned about the serial_number being able to be nil, there is an ems_ref column on canisters can we use physical_storage + ems_ref as the manager_ref and it is up to the lenovo parser to make sure it is present?  I don't want to be solving any of the lenovo specific problems in the generic physical infra inventory collections.

Reply:
there is no `ems_ref` (UUID) for canisters in this Lenovo API request. It's the same problem like in old refresh. There should be called `/canisters` separatedly. But this PR should be as similar as possible as old refresh and another usage of API should be implemented in another PR (..and I have only VCR cassettes where this request isn't available)